modules/secrets-store-google.adoc

```diff
@@ -0,0 +1,326 @@
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-gcp-provider                        <1>
+  namespace: my-namespace                      <2>
+spec:
+  provider: gcp                                <3>
+  parameters:                                  <4>
+    secrets: |
+      - resourceName: "projects/my-project/secrets/testsecret1/versions/1"
+        path: "testsecret1.txt"
+----
+<1> Specify the name for the secret provider class.
+<2> Specify the namespace for the secret provider class.
+<3> Specify the provider as `gcp`.
+<4> Specify the provider-specific configuration parameters.
++
+////
+Changed:
+- app-secrets -> my-gcp-provider
+- testropatil -> my-namespace
+- openshift-qe -> my-project
+////
+
+.. Create the `SecretProviderClass` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f secret-provider-class-gcp.yaml
+----
+
+. Create a deployment to use this secret provider class:
+
+.. Create a YAML file that defines the `Deployment` object:
++
+.Example `deployment.yaml`
+[source,yaml]
+----
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-gcp-deployment                              <1>
+  namespace: my-namespace                              <2>
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: my-storage
+  template:
+    metadata:
+      labels:
+        app: my-storage
+    spec:
+      serviceAccountName: my-service-account           <3>
+      containers:
+      - name: busybox
+        image: k8s.gcr.io/e2e-test-images/busybox:1.29
+        command:
+          - "/bin/sleep"
+          - "10000"
+        volumeMounts:
+        - name: secrets-store-inline
+          mountPath: "/mnt/secrets-store"
+          readOnly: true
+      volumes:
+        - name: secrets-store-inline
+          csi:
+            driver: secrets-store.csi.k8s.io
+            readOnly: true
+            volumeAttributes:
+              secretProviderClass: "my-gcp-provider"   <4>
+            nodePublishSecretRef:
+              name: secrets-store-creds                <5>
+----
+<1> Specify the name for the deployment.
+<2> Specify the namespace for the deployment. This must be the same namespace as the secret provider class.
+<3> Specify the service account you created.
+<4> Specify the name of the secret provider class.
+<5> Specify the name of the Kubernetes secret that contains the service principal credentials to access Google Secret Manager.
++
+////
+- Changed mypodserviceaccount -> my-service-account
+////
++
+// TODO: Confirm nodePublishSecretRef is required
+
+.. Create the `Deployment` object by running the following command:
++
+[source,terminal]
+----
+$ oc create -f deployment.yaml
+----
+
+.Verification
+
+* Verify that you can access the secrets from Google Secret Manager in the pod volume mount:
+
+.. List the secrets in the pod mount:
++
+[source,terminal]
+----
+$ oc exec busybox-<hash> -n my-namespace -- ls /mnt/secrets-store/
```

Comment:
Plz update to oc exec my-gcp-deployment-xxxx -n my-namespace -- ls /mnt/secrets-store/

Reply:
@ropatil010 Just double checking that this is correct? With my-gcp-deployment being the name of the deployment, but the container is named busybox. Looking at the existing AWS examples, we have busybox listed in these steps.

Reply:
Hi Sorry, then better we need to update the doc in those steps as well i guess. 
It always takes podname/depname not container name actually. 
ex: 
dep.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: mydep-fs
    spec:
      containers:
      - name: hello-storage

oc exec hello-storage -n testropatil -- ls /mnt/storage 
Error from server (NotFound): pods "hello-storage" not found

oc exec mydep-fs-686596d4f7-wwrgj -n testropatil -- date            
Thu Sep 26 14:07:22 UTC 2024

@dobsonj Can you plz correct here if i'm wrong?

Reply:
@ropatil010 is right, `oc exec` should reference the pod name, not the container name. So it should be `my-gcp-deployment-<hash>` instead of `busybox-<hash>`. Same goes for the other providers, it should match the pod name.